### PR TITLE
solution

### PR DESCRIPTION
**Pull request by NikitaKarbazin**
[DEMO LINK](https://nikitakarbazin.github.io/react_redux-list-of-posts/)

### CODE REVIEW

**Review by illavodonis — COMMENTED** [review]
Gj :fire:

**Review by illavodonis — APPROVED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -8,52 +8,41 @@ import { PostsList } from './components/PostsList';
...
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+  const { author } = useAppSelector(state => state.users);
+  const {
+    posts,
+    loaded,
+    selectedPost,
+    hasError,
+  } = useAppSelector(state => state.posts);
```
Comment (illiarerun) [inline src/App.tsx:25]: write a selector for each entity, follow DRY, you need to have like `authorSelector`, `postsSelector`

File: src/components/PostDetails.tsx [hunk 2]
```diff
@@ -1,34 +1,39 @@
...
+  // function loadComments() {
+  //   setLoaded(false);
+  //   setError(false);
+  //   setVisible(false);
+  //
+  //   commentsApi.getPostComments(post.id)
+  //     .then(setComments) // save the loaded comments
+  //     .catch(() => setError(true)) // show an error when something went wrong
+  //     .finally(() => setLoaded(true)); // hide the spinner
+  // }
```
Comment (illiarerun) [inline src/components/PostDetails.tsx:30]: don't leave commented code
```suggestion
```

File: src/components/PostDetails.tsx [hunk 3]
```diff
@@ -1,34 +1,39 @@
...
+  setVisibleComments,
+} from '../store/reducers/commentsSlice';
+
+export const PostDetails: React.FC = () => {
+  const {
+    comments,
+    loaded,
+    hasError,
+    visibleComments,
+  } = useAppSelector(state => state.comments);
```
Comment (illiarerun) [inline src/components/PostDetails.tsx:18]: `commentsSelector`

File: src/components/PostDetails.tsx [hunk 4]
```diff
@@ -1,34 +1,39 @@
...
+  const dispatch = useAppDispatch();
+  const { selectedPost } = useAppSelector(state => state.posts);
```
Comment (illiarerun) [inline src/components/PostDetails.tsx:20]: `postsSelector`

File: src/store/reducers/commentsSlice.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        state.comments = action.payload;
+        state.loaded = true;
+      })
+      .addCase(loadComments.rejected, (state) => {
+        state.hasError = true;
+        state.loaded = true;
+      })
+      .addCase(addComment.rejected, (state) => {
+        state.hasError = true;
+      });
```
Comment (illiarerun) [inline src/store/reducers/commentsSlice.ts:77]: `loaded: true`

File: src/store/reducers/commentsSlice.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+      state.visibleComments = !state.visibleComments;
+    },
+    addCommentLocal: (state, action) => {
+      state.comments.push({ ...action.payload, id: Math.random() });
+    },
+    deleteCommentLocal: (state, action) => {
+      state.comments = state
+        .comments
+        .filter(comment => comment.id !== action.payload);
+    },
```
Comment (illiarerun) [inline src/store/reducers/commentsSlice.ts:59]: Do this in your extraReducers, on fullfilment, like `deleteComment.fulfilled, ... =>`

File: src/components/PostsList.tsx [hunk 7]
```diff
@@ -1,55 +1,54 @@
...
+              <th>{post.id}</th>
+              <td>{post.title}</td>
+              <td className="has-text-right is-vcentered">
+                <button
+                  type="button"
+                  className={classNames(
+                    'button',
+                    'is-link',
+                    {
+                      'is-light': post.id !== selectedPostId,
```
Comment (illiarerun) [inline src/components/PostsList.tsx:36]: you repeat this check 3 times, move it to variable

**Review by illiarerun — CHANGES_REQUESTED** [review]

**Pull request closed without merging**